core/tx_pool.go

```diff
@@ -339,17 +341,6 @@ func (pool *TxPool) Pending() (map[common.Address]types.Transactions, error) {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
```

Comment:
That copy shouldn't take too much time imho. It's a few MB max, in memory copy shouldn't cause the mutex to be held too long.